client-lib/src/connector/timeplus/TimeplusSink.ts

```diff
@@ -0,0 +1,353 @@
+    Parameter,
+    SinkState,
+    Schema,
+    UpdateMethod,
+    SinkStateKey,
+    SchemaIdentifier,
+    RecordStreamContext,
+    ParameterType,
+    DPMRecord
+} from "datapm-lib";
+import { JSONSchema7TypeName } from "json-schema";
+import { Transform } from "stream";
+import { JobContext } from "../../task/Task";
+import { Maybe } from "../../util/Maybe";
+import { StreamSetProcessingMethod } from "../../util/StreamToSinkUtil";
+import { CommitKey, Sink, SinkSupportedStreamOptions, WritableWithContext } from "../Sink";
+import { getAuthToken } from "./TimeplusConnector";
+import { DISPLAY_NAME, TYPE } from "./TimeplusConnectorDescription";
+import { fetch } from "cross-fetch";
+import { SemVer } from "semver";
+import { BatchingTransform } from "../../transforms/BatchingTransform";
+
+type TimeplusColumn = {
+    name: string;
+    type: string;
+};
+
+type TimeplusColumns = Array<TimeplusColumn>;
+
+type TimeplusStream = {
+    name: string;
+    columns: TimeplusColumns;
+};
+
+type ListStreamsResponse = Array<TimeplusStream>;
+
+export class TimeplusSink implements Sink {
+    activeStream: string;
+
+    getType(): string {
+        return TYPE;
+    }
+
+    getDisplayName(): string {
+        return DISPLAY_NAME;
+    }
+
+    isStronglyTyped(): boolean | Promise<boolean> {
+        return true;
+    }
+
+    async getParameters(
+        catalogSlug: string | undefined,
+        packageFile: PackageFile,
+        connectionConfiguration: DPMConfiguration,
+        credentialsConfiguration: DPMConfiguration,
+        configuration: DPMConfiguration,
+        jobContext: JobContext
+    ): Promise<Parameter[]> {
+        configuration.schemaStreamNames = {};
+
+        for (const schema of packageFile.schemas) {
+            if (schema.title == null) throw new Error("Schema has no title");
+
+            const schemaReference: SchemaIdentifier = {
+                registryUrl: "",
+                catalogSlug: catalogSlug as string,
+                packageSlug: packageFile.packageSlug as string,
+                majorVersion: new SemVer(packageFile.version).major,
+                schemaTitle: schema.title as string
+            };
+
+            const TimeplusStreamName =
+                schemaReference.catalogSlug +
+                "_" +
+                schemaReference.packageSlug +
+                "_" +
+                schemaReference.majorVersion +
+                "_" +
+                schemaReference.schemaTitle;
+
+            if (configuration["stream-name-" + schema.title] == null) {
+                return [
+                    {
+                        type: ParameterType.Text,
+                        name: "stream-name-" + schema.title,
+                        configuration,
+                        message: "Stream for " + schema.title + " records?",
+                        defaultValue: TimeplusStreamName
+                    }
+                ];
+            }
+
+            await this.getOrCreateStream(
```

Comment:
Discovered a bug in the DecodableSink today where this call should be in the getWritable(...) method... which is called after the user has the option to "deconflict" types. 

I can make this change if you like.